src/librustc_mir/borrow_check/mod.rs

```diff
@@ -1654,7 +1662,7 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
 
                     if let Some((err_help_span, err_help_stmt, item_msg, sec_span)) = err_info {
                         let mut err = self.tcx.cannot_assign(span, &item_msg, Origin::Mir);
-                        err.span_suggestion(err_help_span, err_help_stmt, format!(""));
+                        err.span_suggestion(err_help_span, &err_help_stmt, format!(""));
```

Comment:
If you look [below](https://github.com/rust-lang/rust/pull/49931/files#diff-6ec61eb4b2eddffb0a366aeb761a579eR8), the third argument is the code content that will be suggested. Because here an empty `String` is being passed, the suggestion effectively tells the user to remove the code.